app/src/ereferrals/java/org/eyeseetea/malariacare/strategies/LoginActivityStrategy.java

```diff
@@ -127,6 +128,19 @@ public void onClick(View view) {
                 onForgotPassword();
             }
         });
+        IMainExecutor mainExecutor = new UIThreadExecutor();
+        IAsyncExecutor asyncExecutor = new AsyncExecutor();
+        ICredentialsRepository credentialsRepository = new CredentialsLocalDataSource();
+        GetUserLastInsertedUsernameUseCase getUserLastInsertedUsernameUseCase =
```

Comment:
@manuelplazaspalacio I'm asking here the opinion from @xurxodev because I may be too tired but I wouldn't use a UseCase for this, but directly access the credentialsRepository with:
```
username.setText(credentialsRepository.getOrganisationCredentials().getUsername());
```
The dependency here is going towards "the kernel of the onion", so I guess it's not a problem...but @xurxodev may correct me saying that we need to use a UseCase for accessing the entities in any case...let's wait for his confirmation. In any case, does it make sense for you?

Reply:
@manuelplazaspalacio @ifoche Yes we should to use a Use Case for retrieve data from database because it could entail additional application business rules that would be encapsulated in the use case. But a use case should not return primitive values, it is a code smell must return entities always. I have reviewed the code we already have in ereferrrals a use case that can serve you for what you need GetUserUserAccountUseCase that returns UserAccount and contains username